### DISCUSSION TITLE
The wordwise contents are wrong.

### DISCUSSION THREAD

**Discussion by Garcia-netizen** [xxyzz/WordDumb#123] [Q&A]
I use the plugin to generate the english to english wordwise. The contents show in my kindle, but the words it hints totally wrong, I don't know why, it's so strange.
![avatar](https://i.328888.xyz/2023/05/05/iTAxvv.png)

**Comment by xxyzz** [comment]
Please provide more detailed reproduce steps and also post your plugin and Kindle configurations.

**Reply by Garcia-netizen** [reply]
This is the plugin configurations, there are always errors when I use the generete wordwise and x-ray, so I only use generating wordwise, the language is Chinese I hope it don't affect you to understand. 
![image](https://user-images.githubusercontent.com/103917069/236390087-5c764a3d-8532-4024-8b88-144214bfa444.png)
![image](https://user-images.githubusercontent.com/103917069/236392441-99ba9267-8681-4612-98a6-2545fd9a17c2.png)

My kindle is kp4 and the version is Kindle 5.15.1.1 (392498 005).

**Reply by Garcia-netizen** [reply]
The books are all I send by email before, I connect the kindle with calibre when I use the plugin to generate wordwise, then I disconnect the kindle, my kindle is in aeroplane mode through the process, the wordwise show in books but all hint wrong word. I try to connect wifi after that, but it make no change.

**Reply by xxyzz** [reply]
Books sent via Email are changed by Amazon server, thus the word positions also changed. Delete your books from Kindle then send the book to device through the WordDumb plugin. And could you post the X-Ray error?

**Reply by Garcia-netizen** [reply]
Well, I delete the book and send it through the plugin, but the wordwise is still wrong, this time I can use the X-ray successful, I am not sure  the reason that  errors appeared before.

**Reply by xxyzz** [reply]
Could you upload the book and the generated Word Wise database file?

**Reply by Garcia-netizen** [reply]
I choose one zipped. 
[Rowling, J.K_.zip](https://github.com/xxyzz/WordDumb/files/11407629/Rowling.J.K_.zip)

**Reply by xxyzz** [reply]
I tested the book and I compared the generated Word Wise file with yours and they have many differences, and I couldn't reproduce the bug. Are you using the latest plugin?

The "correct" Word Wise file: [LanguageLayer.en.B01JRE0IU6.kll.zip](https://github.com/xxyzz/WordDumb/files/11414209/LanguageLayer.en.B01JRE0IU6.kll.zip)

**Reply by Garcia-netizen** [reply]
You can use the plugin to generate the correct wordwise? I used my old kindle to try, it is still wrong. I download the plugin in calibre and the version is 3.29.1. 
I use the correct file to replace the wrong in kindle's folder? I have thirties books to genenrate, maybe it isn't easy to do like this one by one. Actually I read the readme but I don't understand configurations of the plugin very well,  I just use it freely.

**Reply by xxyzz** [reply]
The kll file I posted above was created from the test plugin, you can download the test version from GitHub Actions Artifacts: https://github.com/xxyzz/WordDumb/actions/workflows/tests.yml   But I don't think the code changes would affect word wise generation.

You could replace the kll file on your Kindle and see if it works. I don't know why the word locations are incorrect on your machine, maybe your plugin settings is not English to English when creating the file.

**Reply by Garcia-netizen** [reply]
Well, I unchecked "Use Wiktionary definition" in "Choose language" box, it can generate the correct wordwise, I don't know why and wish it can be helpful for you to find the issue.

**Reply by xxyzz** [reply]
You have to select "Chinese" language on Kindle's Word Wise setting to view Wiktionary definition. Please read the plugin usage document: https://xxyzz.github.io/WordDumb/usage.html
